sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/DefaultInputFile.java

```diff
@@ -98,26 +102,45 @@ public String contents() throws IOException {
     }
   }
 
-  /**
-   * @since 6.3
-   */
-  public DefaultInputFile setPublish(boolean publish) {
-    this.publish = publish;
+  public DefaultInputFile setPublished(boolean published) {
+    this.published = published;
     return this;
   }
 
-  /**
-   * @since 6.3
-   */
-  public boolean publish() {
-    return publish;
+  public boolean isPublished() {
+    return published;
+  }
+
+  public DefaultInputFile setExcludedForCoverage(boolean excludedForCoverage) {
+    this.excludedForCoverage = excludedForCoverage;
+    return this;
+  }
+
+  public boolean isExcludedForCoverage() {
+    return excludedForCoverage;
   }
 
+  /**
+   * @deprecated since 6.6
+   */
+  @Deprecated
   @Override
   public String relativePath() {
     return indexedFile.relativePath();
   }
 
+  public String getModuleRelativePath() {
+    return indexedFile.getModuleRelativePath();
+  }
+
+  public String getProjectRelativePath() {
+    return indexedFile.getProjectRelativePath();
+  }
+
+  /**
+   * @deprecated since 6.6
+   */
+  @Deprecated
   @Override
   public String absolutePath() {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)